### PR TITLE
To use acquire_token_interactive from msal

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_credentials/device_code.py [hunk 1]
```diff
@@ -64,28 +64,8 @@ def _request_token(self, *scopes: str, **kwargs) -> Dict:
...
-        if self._timeout is not None and self._timeout < flow["expires_in"]:
-            # user specified an effective timeout we will observe
-            deadline = int(time.time()) + self._timeout
-            result = app.acquire_token_by_device_flow(
-                flow, exit_condition=lambda flow: time.time() > deadline, claims_challenge=kwargs.get("claims")
-            )
-        else:
-            # MSAL will stop polling when the device code expires
-            result = app.acquire_token_by_device_flow(flow, claims_challenge=kwargs.get("claims"))
+        result = app.acquire_token_interactive(scopes=scopes, claims_challenge=kwargs.get("claims"))
```
Comment (rayluo) [inline sdk/identity/azure-identity/azure/identity/_credentials/device_code.py:68]: As explained offline, please revert the change to this device_code.py

File: sdk/identity/azure-identity/azure/identity/_credentials/browser.py [hunk 2]
```diff
@@ -2,18 +2,13 @@
...
-import subprocess
-import webbrowser
 from typing import Dict
 from urllib.parse import urlparse
 
 from azure.core.exceptions import ClientAuthenticationError
 
-from .. import CredentialUnavailableError
 from .._constants import DEVELOPER_SIGN_ON_CLIENT_ID
-from .._internal import AuthCodeRedirectServer, InteractiveCredential, wrap_exceptions
```
Comment (rayluo) [inline sdk/identity/azure-identity/azure/identity/_credentials/browser.py:16]: You probably want to also remove the AuthCodeRedirectServer from inside internal

**Review by pvaneck — COMMENTED** [review]
Looks like `test_browser_credential.py` needs updating due to removal of `AuthCodeRedirectServer`

File: sdk/identity/azure-identity/azure/identity/_credentials/browser.py [hunk 3]
```diff
@@ -56,78 +53,30 @@ def __init__(self, **kwargs) -> None:
...
-            except Exception:  # pylint:disable=broad-except
-                # powershell.exe isn't available, or the subprocess timed out
-                pass
-    return opened
+        except socket.error:
+            raise CredentialUnavailableError(message="Couldn't start an HTTP server.")
+        if "access_token" not in result and "error_description" in result:
+            raise ClientAuthenticationError(message=result.get("error_description"))
+        if "access_token" not in result:
+            raise ClientAuthenticationError(message="Failed for user to authenticate")
```
Comment (pvaneck) [inline sdk/identity/azure-identity/azure/identity/_credentials/browser.py:79]: nit:
```suggestion
            raise ClientAuthenticationError(message="Failed to authenticate user")
```

File: sdk/identity/azure-identity/azure/identity/_credentials/browser.py [hunk 4]
```diff
@@ -56,78 +53,30 @@ def __init__(self, **kwargs) -> None:
...
-        if not response:
-            raise ClientAuthenticationError(
-                message="Timed out after waiting {} seconds for the user to authenticate".format(self._timeout)
+        port = self._parsed_url.port if self._parsed_url else None
+
+        try:
+            result = app.acquire_token_interactive(
+                scopes=scopes,
+                login_hint=self._login_hint,
+                claims_challenge=claims,
```
Comment (pvaneck) [inline sdk/identity/azure-identity/azure/identity/_credentials/browser.py:70]: Should we also propagate timeout?
```suggestion
                claims_challenge=claims,
                timeout=self._timeout,
```

Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/_credentials/browser.py:70]: Though after testing the timeout, the timeout works, but msal doesn't seem to give a proper exception/error message that indicates a timeout...
```
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/application.py", line 1828, in acquire_token_interactive
    response = _clean_up(self.client.obtain_token_by_browser(
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/oauth2cli/oidc.py", line 281, in obtain_token_by_browser
    return super(Client, self).obtain_token_by_browser(
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/oauth2cli/oauth2.py", line 639, in obtain_token_by_browser
    return self._obtain_token_by_browser(
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/oauth2cli/oauth2.py", line 677, in _obtain_token_by_browser
    return self.obtain_token_by_auth_code_flow(
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/application.py", line 134, in obtain_token_by_auth_code_flow
    assert isinstance(auth_code_flow, dict) and isinstance(auth_response, dict)
AssertionError
```

File: sdk/identity/azure-identity/tests/test_browser_credential.py [hunk 5]
```diff
@@ -11,11 +11,9 @@
...
 from azure.core.pipeline.policies import SansIOHTTPPolicy
 from azure.core.pipeline.transport import RequestsTransport
-from azure.identity import AuthenticationRequiredError, CredentialUnavailableError, InteractiveBrowserCredential
-from azure.identity._internal import AuthCodeRedirectServer
+from azure.identity import InteractiveBrowserCredential
 from azure.identity._internal.user_agent import USER_AGENT
 import pytest
-from six.moves import urllib
 
 from helpers import (
```
Comment (pvaneck) [inline sdk/identity/azure-identity/tests/test_browser_credential.py:18]: nit: Can remove all these unused dependencies from helpers since these are no longer used.

**Review by pvaneck — APPROVED** [review]

File: sdk/identity/azure-identity/azure/identity/_credentials/browser.py (hunk 4 above)
Comment (rayluo) [reply sdk/identity/azure-identity/azure/identity/_credentials/browser.py:70]: @pvaneck Thanks for your finding. This is a bug in MSAL. Interesting that nobody reported it before, so I guess nobody really uses the "timeout" behavior.

I'll fix it soon, probably by throwing something like [TimeoutError](https://docs.python.org/3/library/exceptions.html#TimeoutError).

CC: @xiangyan99

Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_credentials/browser.py:70]: @rayluo what is current error?

You may need to evaluate if it is a breaking change.

Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/_credentials/browser.py:70]: Thanks @rayluo. I opened an issue just recently to track it: https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/516
